src/chmatch.c

```diff
@@ -24,26 +24,19 @@ static SEXP chmatchMain(SEXP x, SEXP table, int nomatch, bool chin, bool chmatch
   int *ansd = INTEGER(ans);
   if (!length(table)) { const int val=(chin?0:nomatch), n=LENGTH(x); for (int i=0; i<n; ++i) ansd[i]=val; UNPROTECT(1); return ans; }
   savetl_init();
-  const SEXP *xd = STRING_PTR(x);
   const int xlen = length(x);
+  int nprotect = 1;
+  SEXP ux = x;
   for (int i=0; i<xlen; i++) {
-    SEXP s = xd[i];
-    if (s != NA_STRING && ENC_KNOWN(s) != 64) { // PREV: s != NA_STRING && !ENC_KNOWN(s) - changed to fix for bug #5159. The previous fix
```

Comment:
helpful for future readers if you leave these comments and then write your own explaining the shortfalls

Reply:
Thanks, added.